ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Abstract.hs

```diff
@@ -102,24 +113,76 @@ class ApplyBlock (LedgerState blk) blk => UpdateLedger blk
   Derived functionality
 -------------------------------------------------------------------------------}
 
-tickThenApply :: ApplyBlock l blk
-              => LedgerCfg l -> blk -> l -> Except (LedgerErr l) l
-tickThenApply cfg blk =
-      applyLedgerBlock cfg blk
-    . applyChainTick cfg (blockSlot blk)
-
-tickThenReapply :: ApplyBlock l blk
-                => LedgerCfg l -> blk -> l -> l
-tickThenReapply cfg blk =
-      reapplyLedgerBlock cfg blk
-    . applyChainTick cfg (blockSlot blk)
-
-foldLedger :: ApplyBlock l blk
-           => LedgerCfg l -> [blk] -> l -> Except (LedgerErr l) l
+-- | 'lrResult' after 'applyBlockLedgerResult'
+applyLedgerBlock ::
+     (ApplyBlock l blk, HasCallStack)
+  => LedgerCfg l
+  -> blk
+  -> Ticked l
+  -> Except (LedgerErr l) l
+applyLedgerBlock = fmap lrResult ..: applyBlockLedgerResult
+
+-- | 'lrResult' after 'reapplyBlockLedgerResult'
+reapplyLedgerBlock ::
+     (ApplyBlock l blk, HasCallStack)
+  => LedgerCfg l
+  -> blk
+  -> Ticked l
+  -> l
+reapplyLedgerBlock = lrResult ..: reapplyBlockLedgerResult
+
+tickThenApplyLedgerResult ::
+     ApplyBlock l blk
+  => LedgerCfg l
+  -> blk
+  -> l
+  -> Except (LedgerErr l) (LedgerResult l l)
+tickThenApplyLedgerResult cfg blk l = do
+  let lrTick = applyChainTickLedgerResult cfg (blockSlot blk) l
+  lrBlock <-   applyBlockLedgerResult     cfg            blk  (lrResult lrTick)
+  pure LedgerResult {
+      lrEvents = lrEvents lrTick <> lrEvents lrBlock
```

Comment:
Confirm that left-to-right = first-to-later.

Reply:
yes, that is the case